### PR TITLE
Chore: `no-control-regex`: Remove seemingly unnecessary logic.

### PR DESCRIPTION
**Pull request by ljharb**
While investigating a fix for #6293, I discovered that the test in https://github.com/eslint/eslint/commit/141b77841e1c086d2f1b73e767d2fe3be4590a17 continues to pass even without this code.

Either this code should be deleted, or tests should be added that necessitate its existence. I'm not sure which, but it seemed like the former was correct.

/cc @efegurkan

### CODE REVIEW

**Comment by nzakas** [comment]
Hmmm, removing that code basically reverts 141b778, so something seems off here. I'm willing to bet that tests are missing.  @efegurkan can you confirm?

**Comment by ljharb** [comment]
Yes, I agree - I'll be refactoring all this code regardless to fix #6293, so having the tests would greatly help me avoid breaking things :-)

**Comment by efegurkan** [comment]
Hello @ljharb, @nzakas,

**TL;DR** Code is necessary but tests are missing as you mentioned. I will try to add tests on weekend.

So when you declare a new regex with regex literal like ``` var regex = /\x1f/ ``` it has control characters in it and run toString on it, it outputs ```/\x1f/``` and the output does not have control characters in it BUT, regex itself has control characters. So the regex match on the rule was not working.

When you declare it as ```new Regex('\x1f')``` toString output has control characters in it and regex match on the rule.

@ljharb so you can keep that in mind while refactoring the code to make sure you don't break anything until tests are added.

**Comment by nzakas** [comment]
Thanks @efegurkan. You can reference issue #6438 when you submit that change. 

@ljharb are you good with closing this?

**Comment by ljharb** [comment]
Sure, I'll close this - but the tests already contain a regex literal with control characters, so I'm not seeing the behavior @efegurkan is describing. I look forward to the tests :-)

**Comment by efegurkan** [comment]
@ljharb as you mentioned tests contains a literal which should have failed when we deleted code between line 65-75. But rule itself fails when you do this even if tests are passing.

I believe there is something with test execution going on here, I am trying to find out at the moment but I might need some assistance later on.
